ql/src/test/org/apache/hadoop/hive/ql/TestTxnExIm.java

```diff
@@ -389,21 +390,56 @@ public void testImportPartitioned() throws Exception {
 
   @Test
   public void testImportPartitionedOrc() throws Exception {
+    //clear and drop table T,Tstage
     runStatementOnDriver("drop table if exists T");
     runStatementOnDriver("drop table if exists Tstage");
-    runStatementOnDriver("create table T (a int, b int) partitioned by (p int) stored" +
-        " as orc tblproperties('transactional'='true')");
-    //Tstage is the target table
-    runStatementOnDriver("create table Tstage (a int, b int) partitioned by (p int) stored" +
-        " as orc tblproperties('transactional'='true')");
+
+    //create target table - T
+    runStatementOnDriver("create table T (a int, b int) partitioned by (p int) stored"
+        + " as orc tblproperties('transactional'='true')");
+
+    //create source table - Tstage
+    runStatementOnDriver("create table Tstage (a int, b int) partitioned by (p int) stored"
+        + " as orc tblproperties('transactional'='true')");
+
     //this creates an ORC data file with correct schema under table root
     runStatementOnDriver("insert into Tstage values(1,2,10),(3,4,11),(5,6,12)");
-    final int[][] rows = {{3}};
-    //now we have an archive with 3 partitions
+    final int[][] rows = { { 3 } };
+
+    //check Partitions statistics
+    List<String> rsTstagePartitionsProperties = runStatementOnDriver("show partitions Tstage");
+    for (String rsTstagePartition : rsTstagePartitionsProperties) {
+      List<String> rsPartitionProperties =
+          runStatementOnDriver("describe formatted Tstage partition(" + rsTstagePartition + ")");
+      Assert.assertEquals("COLUMN_STATS_ACCURATE of partition " + rsTstagePartition + " of Tstage table", true,
+          rsPartitionProperties.contains("\tCOLUMN_STATS_ACCURATE\t{\\\"BASIC_STATS\\\":\\\"true\\\"}"));
+      Assert.assertEquals(" of partition " + rsTstagePartition + " of Tstage table", true,
+          rsPartitionProperties.contains("\tnumRows             \t1                   "));
+    }
+
+    //now we have an archive Tstage with 3 partitions
     runStatementOnDriver("export table Tstage to '" + getWarehouseDir() + "/1'");
 
     //load T
     runStatementOnDriver("import table T from '" + getWarehouseDir() + "/1'");
+
+    //check basic stats in tblproperties of T
+    List<String> rsTProperties = runStatementOnDriver("show tblproperties T");
+    Assert.assertEquals("COLUMN_STATS_ACCURATE of T table", false,
```

Comment:
I think, stats are false as table T was already created. Can we check if it is set to false when import create the table.